app/src/main/java/com/example/househomey/filter/model/TagFilter.java

```diff
@@ -0,0 +1,42 @@
+package com.example.househomey.filter.model;
+
+import com.example.househomey.Item;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A filter class that filters a list of items based on selected tags.
+ * @author Jared Drueco
+ */
+public class TagFilter extends Filter {
+    public Map<String, Boolean> tagSelectionMap;
+
+    /**
+     * Constructs a new TagFilter with the specified keywords.
+     *
+     * @param tagSelectionMap tags to filter by.
+     */
+    public TagFilter(Map tagSelectionMap) {
+        this.tagSelectionMap = tagSelectionMap;
+    }
+
+    /**
+     * Filters the passed in list of items, retaining only those items that match the
+     * tags specified in this filter.
+     *
+     * @param itemList The original list of items to be filtered.
+     * @return A filtered list of items that have at least one of the tags.
+     */
+    @Override
+    public ArrayList<Item> filterList(ArrayList<Item> itemList) {
+        if (!tagSelectionMap.values().stream().anyMatch(Boolean::booleanValue)) return itemList;
```

Comment:
why not just ``` if (tagSelectionMap.isEmpty())```

Reply:
this does not take into account the case where there are tags in the map but they are all false. it will result in the filter still filtering by an empty tag